nosepipe.py

```diff
@@ -147,12 +147,16 @@ def __call__(self, result):
             useshell = True
 
         self.logger.debug("Executing %s", " ".join(argv))
-        popen = subprocess.Popen(argv,
-                                 cwd=self._cwd,
-                                 stdout=subprocess.PIPE,
-                                 stderr=subprocess.STDOUT,
-                                 shell=useshell,
-                                 )
+        try:
+            popen = subprocess.Popen(argv,
+                                     cwd=self._cwd,
+                                     stdout=subprocess.PIPE,
+                                     stderr=subprocess.STDOUT,
+                                     shell=useshell,
+                                     )
+        except OSError, e:
```

Comment:
Fixed this as well as the # in comment.  How are you testing?  I’ve tried setup.py nosetests and get lots of problems.


> On Feb 17, 2015, at 2:45 PM, Dan McCombs <notifications@github.com> wrote:
> 
> In nosepipe.py <https://github.com/dmccombs/nosepipe/pull/7#discussion_r24846994>:
> 
> > @@ -147,12 +147,16 @@ def __call__(self, result):
> >              useshell = True
> >  
> >          self.logger.debug("Executing %s", " ".join(argv))
> > -        popen = subprocess.Popen(argv,
> > -                                 cwd=self._cwd,
> > -                                 stdout=subprocess.PIPE,
> > -                                 stderr=subprocess.STDOUT,
> > -                                 shell=useshell,
> > -                                 )
> > +        try:
> > +            popen = subprocess.Popen(argv,
> > +                                     cwd=self._cwd,
> > +                                     stdout=subprocess.PIPE,
> > +                                     stderr=subprocess.STDOUT,
> > +                                     shell=useshell,
> > +                                     )
> > +        except OSError, e:
> Just tried this with Python 3 and noticed this is Python 2 only code - it should instead be except OSError as e:.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/dmccombs/nosepipe/pull/7/files#r24846994>.
>